src/spikeinterface/extractors/KilosortTempWhExtractor.py

```diff
@@ -59,37 +75,168 @@ def __init__(self, output_path: Path) -> None:
-        #      if contact_vector is not None:
-        contact_vector["device_channel_indices"] = np.arange(len(new_channel_ids), dtype="int64")
-        self.set_property("contact_vector", contact_vector)
+    def has_spikeinterface(self, path_: Path) -> bool:
+        """ """
+        sorter_output = path_ / "sorter_output"
 
-        data2 = original_recording.get_traces(start_frame=0, end_frame=75000)
-        breakpoint()
+        if not (path_ / "spikeinterface_recording.json").is_file() or not sorter_output.is_dir():
+            return False
+
+        return self.has_valid_sorter_output(sorter_output)
+
+    def has_valid_sorter_output(self, path_: Path) -> bool:
+        """ """
+        required_files = ["temp_wh.dat", "channel_map.npy", "channel_positions.npy"]
+
+        for filename in required_files:
+            if not (path_ / filename).is_file():
+                print(f"The file {filename} cannot be out in {path_}")
+                return False
+        return True
+
+    def load_and_check_kilosort_params_file(self) -> Dict:
+        """ """
+        params = runpy.run_path(self.sorter_output_path / "params.py")
+
+        if params["dtype"] != "int16":
+            raise ValueError("The dtype in kilosort's params.py is expected" "to be `int16`.")
+
+        return params
 
 
 #        original_probe = original_recording.get_probe()
-# self.set_probe(original_probe)
+# self.set_probe(original_probe) TODO: do we need to adjust the probe? what about contact positions?
 
 # 1) figure out metadata and casting for WaveForm Extractor
 # 2) check lazyness etc.
 
 # zero padding can just be kept. Check it plays nice with WaveformExtractor...
 
 # TODO: add provenance
-
+# TODO: what to do about all those zeros?
 
 #    def get_num_samples(self):
 #       """ ignore Kilosort's zero-padding """
 #      return self.original_recording.get_num_samples()
 
-path_ = Path(r"X:\neuroinformatics\scratch\jziminski\ephys\code\sorter_output")
-data = KilosortTempWhExtractor(path_)
+# TODO: check, there must be a probe if sorting was run?
+# change the wiring of the probe
+# TODO: check this carefully, might be completely wrong
+#      if contact_vector is not None:
+
+# if channel_map.ndim == 2:  # kilosort > 2
+#    channel_indices = channel_map.ravel()  # TODO: check multiple shanks
+
+# self.set_channel_locations(new_channel_locations)  # TOOD: check against slice_channels
+
+#             is_filtered=None,  # TODO: need to get from KS provenence?
+
+# In general, do we store the full channel map in channel contacts or do we
+# only save the new subset? My guess is subset for contact_positions, but full probe
+# for probe. Check against slice_channels.
+#             self.set_probe(probe)  # TODO: what does this mean for missing channels?
+
+#         if channel_map.ndim == 2:  # kilosort > 2
+# does kilosort > 2 store shanks differently?             channel_indices = channel_map.ravel()
+
```

Comment:
@JoeZiminski ,

Kilosort  2 no longer uses the shank definition (https://github.com/MouseLand/Kilosort/issues/155), so this is definitely true for Kilosort2. Based on my search of the code base of Kilosort 3 it seems like kcoords (shanks) are still not being used: (https://github.com/search?q=repo%3AMouseLand%2FKilosort%20kcoords&type=code) (ie the code is for loading the kcoords and putting them in rez, but then they aren't actually used.

Reply:
Actually not used in KS2.5 either.
https://github.com/MouseLand/Kilosort/issues/262